pkg/credentialprovider/config.go

```diff
@@ -30,6 +30,13 @@ import (
 	"github.com/golang/glog"
 )
 
+// DockerConfigJson represents ~/.docker/config.json file info
+// see https://github.com/docker/docker/pull/12009
+type DockerConfigJson struct {
+	Auths       DockerConfig      `json:"auths"`
+	HttpHeaders map[string]string `json:"HttpHeaders,omitempty"`
```

Comment:
> @deads2k Can you explain what you mean by "...take the HttpHeaders support as a follow on item... without plumbing them through.".

The `HTTPHeaders` field is used by docker to add opaque headers to client requests.  The new format supports this field, but actually implementing in the kubelet will take additional work.  I'd prefer to avoid having that feature block this pull.  This pull is very useful even if the `HTTPHeaders` field doesn't work as expected.